c10/core/CPUCachingAllocator.cpp

```diff
@@ -91,26 +98,44 @@ CachingAllocatorInfo& GetThreadLocalCachingAllocatorInfo() {
   return caching_allocator_info;
 }
 
-void CachingAllocatorInfo::set(
-    const CachingAllocatorInfo& other) {
-  *this = other;
-}
-
 bool CachingAllocatorInfo::enabled() {
   return is_enabled_;
 }
```

Comment:
What would you think of eliminating `CachingAllocatorInfo` and `enabled` entirely, and just having a thread-local "current allocation cache" pointer?

Reply:
Yeah, I thought about it. I just felt it was cleaner this way to check if you have CachingAllocator enabled. But I dont have a strong preference. I will make the change. I assume you want to do it that way simplify code.